daq/base_gateway.py

```diff
@@ -45,6 +44,10 @@ def __init__(self, runner, name, port_set, env_params=None):
         self.activated = False
         self.result_linger = False
         self.dhcp_monitor = None
+        is_native = bool(self.runner.run_trigger.get('native_vlan'))
+        max_hosts = self.runner.run_trigger.get('max_hosts') or DEFAULT_TEST_HOSTS
+        num_host_ports = max_hosts if is_native else DEFAULT_TEST_HOSTS
```

Comment:
it looks like theres nothing stopping a user to input a huge value here (like 1000) which will eat into the mirror port.

Reply:
I added an assert elsewhere. Turns out there's no easy way to do this given the current structure of the code, since the necessary values are split between topology and gateway. What I did will throw an assertion when the error condition is reached... not ideal, but better than silently just doing something weird.